### PR TITLE
Add Test for Japanese First Year of Era

### PR DESCRIPTION
**Pull request by tarekgh**

### CODE REVIEW

File: src/System.Globalization/tests/DateTimeFormatInfo/DateTimeFormatInfoTests.cs [hunk 1]
```diff
@@ -161,5 +161,25 @@ public void TestHebrewMonths()
...
             }
         }
+
+        [Fact]
+        public void TestFirstYearOfJapaneseEra()
+        {
+            DateTimeFormatInfo jpnFormat = new CultureInfo("ja-JP").DateTimeFormat;
+            jpnFormat.Calendar = new JapaneseCalendar();
+
+            string pattern = "gg yyyy'年' MM'月' dd'日'";
```
[the remark this answers is not in the document]
Comment (tarekgh) [reply src/System.Globalization/tests/DateTimeFormatInfo/DateTimeFormatInfoTests.cs]: I prefer to use the actual character here. if we lose it, the test will fail.

[the remark this answers is not in the document]
Comment (krwq) [reply src/System.Globalization/tests/DateTimeFormatInfo/DateTimeFormatInfoTests.cs]: as long as the test fails I'm fine with this

[the remark this answers is not in the document]
Comment (filipnavara) [reply src/System.Globalization/tests/DateTimeFormatInfo/DateTimeFormatInfoTests.cs]: I was asked couple of times to use \uXXXX syntax and I am convinced it's for a good reason. At least some versions of Visual Studio tend to save the file as UTF-16 when changed and that makes everyone's life miserable.

File: src/System.Globalization/tests/DateTimeFormatInfo/DateTimeFormatInfoTests.cs [hunk 2]
```diff
@@ -168,13 +168,13 @@ public void TestFirstYearOfJapaneseEra()
             DateTimeFormatInfo jpnFormat = new CultureInfo("ja-JP").DateTimeFormat;
             jpnFormat.Calendar = new JapaneseCalendar();
 
-            string pattern = "gg yyyy'年' MM'月' dd'日'";
+            string pattern =  "gg yyyy'\u5E74' MM'\u6708' dd'\u65E5'"; // "gg yyyy'年' MM'月' dd'日'"
```
Comment (Clockwork-Muse) [inline src/System.Globalization/tests/DateTimeFormatInfo/DateTimeFormatInfoTests.cs]: nit: extra leading space

**Comment by tarekgh** [comment]
@dotnet-bot test this please

**Pull request merged**